server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsConfigs.java

```diff
@@ -133,8 +134,8 @@ private static void validateProperties(Properties properties) {
 
         // Make sure that push interval is between 100ms and 1 hour.
         if (properties.containsKey(PUSH_INTERVAL_MS)) {
-            int pushIntervalMs = Integer.parseInt(properties.getProperty(PUSH_INTERVAL_MS));
-            if (pushIntervalMs < MIN_INTERVAL_MS || pushIntervalMs > MAX_INTERVAL_MS) {
+            Integer pushIntervalMs = (Integer) ConfigDef.parseType(PUSH_INTERVAL_MS, properties.getProperty(PUSH_INTERVAL_MS), Type.INT);
```

Comment:
If I get it right, the issue we are trying to solve is regarding the correct exception when config is not as INT or expected? If that's the case then either of the 2 approach I think would be better:

1. Make this method non-static, instance method. And remove `properties` as input. Then instantiate ClientMetricsConfig where they are used and call validate. The downside of this is that validation also happens in `ControllerConfigurationValidator` where unnecessary `ClientMetricsConfig` will be created.
2. Enclose a try/catch in validate method and return what received in exception but can return config exception where exception translates to UnknownServer Exception. This is not very clean but a good way to handle.
3. Use `CONFIG.parse(properties)` in validate method to get `Map<?,?> valueMaps = CONFIG.parse(properties);` and then iterate on map as like we iterate on properties. That shall solve your problem.

I would have chosen option 3. Also if you detail on the prolem satetment in PR description or jira then I can help better. But as far as I think, option 3 might solve your issue.

Reply:
I saw the updated jira. Let me know if option 3 works for you, ideally should.

Reply:
Hi @apoorvmittal10 , sorry for late response.
I have updated this PR according your great suggestion.
Please take a look.